### ISSUE TITLE
[A11Y] Improve the dialog to conform with WGAC

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#5670]
## Description of the bug

Dialogs in Backdrop aren't bad, actually. Most things conform to WGAC requirements.

There are some small things that could get improved, though.

1. When a dialog gets opened with a link (`<a>` tag), the link should have an attribute to signal that: [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup)
2. The dialog "close" button has no text at all, only a title attribute - let's add proper text as "element-invisible", and it shouldn't only be "Close", but actually "Close dialog" to make clear what it does.
3. In the dialog `aria-describedby` points to no actual description - that misleading attribute should be removed
4. The dialog has no [aria-modal](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal) attribute

## Additional infos

What I've learned: screen readers usually do *not* read title attributes, but what they do read aloud (some at least) are css ::before and ::after contents.
